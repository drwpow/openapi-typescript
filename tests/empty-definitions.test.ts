import openapiTS from "../src/index";

describe("allow empty definitions", () => {
  it("allow empty definitions", async () => {
    const schema = {
      swagger: "2.0",
      paths: {
        "/pet": {
          post: {
            description: "",
            operationId: "addPet",
            parameters: [
              {
                in: "body",
                name: "body",
                description: "Pet object that needs to be added to the store",
                required: true,
                schema: {
                  $ref: 'definitions["Pet"]',
                },
              },
            ],
            responses: {
              405: {
                description: "Invalid input",
              },
            },
          },
        },
      },
    };

    // We are using snapshots to enforce consistency in string output and indentation
    expect(await openapiTS(schema as any, { version: 2 })).toMatchSnapshot();
    expect(await openapiTS(schema as any, { immutableTypes: true, version: 2 })).toMatchSnapshot();
  });
});
