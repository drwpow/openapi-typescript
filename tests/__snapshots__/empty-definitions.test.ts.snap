// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allow empty definitions allow empty definitions 1`] = `
"/**
 * This file was auto-generated by openapi-typescript.
 * Do not make direct changes to the file.
 */

export interface paths {
  \\"/pet\\": {
    post: operations[\\"addPet\\"];
  };
}

export interface operations {
  addPet: {
    parameters: {
      body: {
        /** Pet object that needs to be added to the store */
        body: definitions[\\"Pet\\"];
      };
    };
    responses: {
      /** Invalid input */
      405: unknown;
    };
  };
}

export interface external {}
"
`;

exports[`allow empty definitions allow empty definitions 2`] = `
"/**
 * This file was auto-generated by openapi-typescript.
 * Do not make direct changes to the file.
 */

export interface paths {
  readonly \\"/pet\\": {
    readonly post: operations[\\"addPet\\"];
  };
}

export interface operations {
  readonly addPet: {
    readonly parameters: {
      readonly body: {
        /** Pet object that needs to be added to the store */
        readonly body: definitions[\\"Pet\\"];
      };
    };
    readonly responses: {
      /** Invalid input */
      readonly 405: unknown;
    };
  };
}

export interface external {}
"
`;
